### ISSUE TITLE
Add checker for comparing for equality to empty literal

### ISSUE THREAD

**Issue by DanielNoord** [PyCQA/pylint#4774]
### Current problem

_Originally posted by @cdce8p in https://github.com/PyCQA/pylint/pull/4769#discussion_r679096534_

Best not to check for equality by comparing it to an empty literal.
(This should probably be a pylint check as well.)

### Desired solution

```python
# good
test_list = []
if not test_list:
   pass

# bad
test_list = []
if test_list == []:
   pass
```

**Comment by cdce8p** [comment]
I already though of a way to implement it, but couldn't yet think of a good name or description. If anyone once to take a stab at it, this here should work:
```py
    def visit_compare(self, node: astroid.Compare) -> None:
        for op, comparator in node.ops:
            if op == "==":
                if (
                    isinstance(comparator, (astroid.List, astroid.Tuple))
                    and not comparator.elts
                    or isinstance(comparator, astroid.Dict)
                    and not comparator.items
                ):
                    self.add_message(..., node=node)
```

**Comment by DanielNoord** [comment]
Perhaps `no-empty-literal-comparison`?
And what kind of message would this be? Does this have a performance effect? Or is it just code style?

**Comment by cdce8p** [comment]
> Perhaps `no-empty-literal-comparison`?

Sounds like a start. I'm not good at naming things though 😅
@Pierre-Sassoulas Do you have a good idea?

> And what kind of message would this be? Does this have a performance effect? Or is it just code style?

Maybe a small performance effect and a bit code style. So probably a `refactoring` checker.

**Comment by Pierre-Sassoulas** [comment]
Maybe ``use-implicit-booleanness`` or ``empty-litteral-as-condition`` (based on the name of ``len-as-condition`` which is similar) ? This checker is prone to false positive because for numpy/pandas array the boolean value of an array is ambiguous. Ie you need to do ``not test_list.size`` or use ``all`` or ``any``. Arguably ``test_list == []`` could be considered better in that case (?). So I think we probably need to do something specifically for numpy/pandas. Something similar was done for len-as-condition where we check that the ``__bool__`` function is not re-implemented and that the class inherit directly from a list, set, or dict (ie: not a ndarray), see https://github.com/PyCQA/pylint/pull/3821

**Comment by Pierre-Sassoulas** [comment]
I've been thinking: we could rename `len-as-condition` to `use-implicit-booleanness` and add the comparison to empty literal to the checker.

**Comment by DanielNoord** [comment]
Probably should also implement #3031 when working on this issue.

**Comment by jaehoonhwang** [comment]
Hi, can I try to resolve this issue?

**Comment by Pierre-Sassoulas** [comment]
Hi, @jaehoonhwang , of course, I'll assign you to the issue.
